app/src/main/java/cm/aptoide/pt/FirstLaunchManager.kt

```diff
@@ -0,0 +1,81 @@
+package cm.aptoide.pt
+
+import android.content.Context
+import android.content.Intent
+import android.content.SharedPreferences
+import android.content.res.XmlResourceParser
+import cm.aptoide.accountmanager.AptoideAccountManager
+import cm.aptoide.pt.crashreports.CrashReport
+import cm.aptoide.pt.networking.IdsRepository
+import cm.aptoide.pt.root.RootAvailabilityManager
+import cm.aptoide.pt.updates.UpdateRepository
+import cm.aptoide.pt.util.PreferencesXmlParser
+import cm.aptoide.pt.view.MainActivity
+import org.xmlpull.v1.XmlPullParserException
+import rx.Completable
+import java.io.IOException
+import java.lang.Boolean
+
+class FirstLaunchManager(private val defaultSharedPreferences: SharedPreferences,
+                         private val idsRepository: IdsRepository,
+                         private val followedStoresManager: FollowedStoresManager,
+                         private val updateRepository: UpdateRepository,
+                         private val rootAvailabilityManager: RootAvailabilityManager,
+                         private val accountManager: AptoideAccountManager,
+                         private val shortcutManager: AptoideShortcutManager,
+                         private val context: Context) {
+
+  fun runFirstLaunch(): Completable {
+    return Completable.mergeDelayError(setSharedPreferencesValues(), generateAptoideUuid())
+        .andThen(Completable.mergeDelayError(followedStoresManager.setDefaultFollowedStores(),
+            updateRepository.sync(true, false),
+            rootAvailabilityManager.updateRootAvailability(), accountManager.updateAccount(),
+            createShortcut()))
+        .doOnError { e -> CrashReport.getInstance().log(e) }
+  }
+
+  private fun generateAptoideUuid(): Completable {
+    return Completable.fromAction { idsRepository.getUniqueIdentifier() }
```

Comment:
Android studio is giving a warning to use property access syntax

Reply:
Personally, I don't like to use the property access syntax in methods where it's not a simple getter (i.e. it doesn't just return a field). Which is this case, which is why I didn't use it.